### PR TITLE
Add mutiple scope columns, and fix error handling for missing scopes

### PR DESCRIPTION
**Pull request by kriswehner**
The scope columns are optional, and configured with a comma separated
column list on the command line. This also fixes a test that was non-functional
that I wrote some time back, and no longer errors out when there's
no scope column on a given table

### CODE REVIEW

File: common/common.go [hunk 1]
```diff
@@ -16,21 +18,25 @@ type Change struct {
...
-			return c.GetString_(), nil
+func (c *Change) getColumnValue(columns []*proto.Column, names []string) (val string, err error) {
+	for _, name := range names {
+		for _, c := range columns {
+			if c.Name == name {
+				log.Debugf("Found scope with name %s, col is %s", name, c.Name)
+				return c.GetString_(), nil
+			}
+			log.Debugf("Did not find scope with name %s, col is %s", name, c.Name)
 		}
```
Comment (ogrubes) [inline common/common.go]: generalization comment: it looks like getColumnValue is only used in the context of scopes rn... but these methods in common.go look like they're trying to be more general.

with that in mind, (1) I love the debug statements, (2) let's replace "scope" text with "column" in this method?

File: changeserver/server_main_test.go [hunk 2]
```diff
@@ -131,6 +136,7 @@ var _ = AfterSuite(func() {
 	// Drop the test data table
 	executeSQL(dropTableSQL)
 	executeSQL(dropAlternativeScopeTableSQL)
+	executeSQL(dropMutlipleScopeColumnsSQL)
```
Comment (ogrubes) [inline changeserver/server_main_test.go]: grep nit: typo
s/dropMutlipleScopeColumnsSQL/dropMultipleScopeColumnsSQL/g

File: changeserver/changesapi_test.go [hunk 3]
```diff
@@ -49,36 +49,86 @@ var _ = Describe("Changes API", func() {
...
+
+	It("should sync a change from the first configured scope, even if its the different one", func() {
+		testServer.receiver.SetChangeFilter(func(c *common.Change) bool {
+			return c.Table == "public.changeserver_multi_scope_test"
+		})
+		defer func() {
+			testServer.receiver.SetChangeFilter(func(c *common.Change) bool {
+				return c.Table == "public.changeserver_test"
+			})
+		}()
```
Comment (ogrubes) [inline changeserver/changesapi_test.go]: oh these defer functions are super crafty

**Comment by chuckg** [comment]
I don't understand how we'll use multiple scope columns practically and gain anything by it's existence and I think it confuses the API of changeserver pretty dramatically.  Can we split the fix for tables with no scopes and multiple scopes into separate reviews?

File: changeserver/changesapi_test.go (hunk 3 above)
Comment (kriswehner) [inline changeserver/changesapi_test.go]: :). They're almost too crafty, but I love them just the same.

**Comment by kriswehner** [comment]
I'm trying to parse this:
"Can we split the fix for tables with no scopes and multiple scopes into separate reviews?"

Do you mean the stuff to suppress the error logging? It's tiny and not worth it.

File: common/common.go (hunk 1 above)
Comment (kriswehner) [inline common/common.go]: Agreed. This false generalization has bothered me for a long time. So I:
- Moved the function into replicator.go, made it private, acknowledge it's not general.
- Removed the word scope from the logging

I added that logging in anger when I had a bug where the loops were reversed.

File: changeserver/server_main_test.go (hunk 2 above)
Comment (kriswehner) [inline changeserver/server_main_test.go]: Ironically gogland showed me the spelling error and I ignored it. Bad programmer.

File: changeserver/replicator.go [hunk 4]
```diff
@@ -63,12 +64,25 @@ func (s *server) handleChange(c *common.Change, firstChange common.Sequence) (co
...
+}
+
 func getScope(c *common.Change) (string, error) {
 	if c.NewColumns != nil {
-		return c.GetNewColumnValue(scopeFields)
+		return getColumnValue(c.NewColumns, scopeFields)
 	}
 	if c.OldColumns != nil {
-		return c.GetOldColumnValue(scopeFields)
+		return getColumnValue(c.OldColumns, scopeFields)
```
Comment (ogrubes) [inline changeserver/replicator.go]: ahh, this is so much more obvious to place here. nice!

**Pull request merged**